addons/calendar/models/calendar_alarm_manager.py

```diff
@@ -134,6 +134,7 @@ def get_next_mail(self):
 
     @api.model
     def _get_partner_next_mail(self, partners=None):
+        self = self.with_context(mail_notify_force_send=True)
```

Comment:
Why do you really want to send emails now and not use the queue ? Most of automatic or semi-automatic processes in odoo use the mail queue to avoid blocking users.

Reply:
Well, for example if we have 2 events with one reminders each. the first 1 hour before the starting hour and the other 30 minutes before. The cron is called and the first reminder is sent in the queue, if the cron for the reminder are synchronized then the mail is sent almost directly, if not there will be a delay. Then 30 minutes later, the cron for the reminder put the reminder in the queue which will be send 30 minutes later at best because the mailing queue is executed every hour.
So I think that for a calendar reminder system, it should only depends on the frequency of one cron (the reminder one), which can also introduce a delay.

Reply:
I see that _send_mail_to_attendees accepts a force_send argument, propagated to message_notify, and set to True in do_mail_reminder . Is your context key still necessary ? Normally it is already in force send mode ...